src/fsharp/TypeChecker.fs

```diff
@@ -15030,7 +15030,9 @@ module EstablishTypeDefinitionCores =
             let hasMeasureableAttr = HasFSharpAttribute cenv.g cenv.g.attrib_MeasureableAttribute attrs
             let hasCLIMutable = HasFSharpAttribute cenv.g cenv.g.attrib_CLIMutableAttribute attrs
             
-            let hasStructLayoutAttr = HasFSharpAttribute cenv.g cenv.g.attrib_StructLayoutAttribute attrs
+            let structLayoutAttr = TryFindFSharpInt32Attribute cenv.g cenv.g.attrib_StructLayoutAttribute attrs
+            let hasStructLayoutAttr = Option.isSome structLayoutAttr
```

Comment:
Can you please move these things down to usage. Maybe then you can get rid of isSome by using pattern matching